packages/atlas/src/CommonProviders.tsx

```diff
@@ -13,12 +17,37 @@ export const CommonProviders: FC<PropsWithChildren> = ({ children }) => {
   return (
     <>
       <GlobalStyles />
-      <ApolloProvider client={apolloClient}>
-        <OperatorsContextProvider>
-          <AssetsManager />
-          <BrowserRouter>{children}</BrowserRouter>
-        </OperatorsContextProvider>
-      </ApolloProvider>
+      <OverlayManagerProvider>
+        <ApolloProvider client={apolloClient}>
+          <MaintenanceWrapper>
+            <OperatorsContextProvider>
+              <AssetsManager />
+              <BrowserRouter>{children}</BrowserRouter>
+            </OperatorsContextProvider>
+          </MaintenanceWrapper>
+        </ApolloProvider>
+      </OverlayManagerProvider>
+    </>
+  )
+}
+
+const MaintenanceWrapper: FC<PropsWithChildren> = ({ children }) => {
+  const { isKilled, wasKilledLastTime, error, loading } = useGetKillSwitch()
+
+  return (
+    <>
+      {isKilled || (error && wasKilledLastTime) || (loading && wasKilledLastTime) ? (
+        <>
+          <Maintenance />
+        </>
+      ) : (
+        children
+      )}
```

Comment:
Let's use regular `if` conditions and `return` statements for this, it's hard to read atm

Reply:
It won't be possible because we need to render `AdminModal` conditionally. I mean it would be possible but readability wouldn't be better. I cleaned up it a little bit, let me know if this is fine for you.